web/src/engine/websites/Bomtoon.ts

```diff
@@ -37,4 +61,53 @@ export default class extends Delitoon {
         return results.distinct();
     }
 
+    public override async FetchPages(chapter: Chapter): Promise<Page<ScrambleParams>[]> {
+        await this.UpdateToken();
+        const url = new URL(`contents/viewer/${chapter.Parent.Identifier}/${chapter.Identifier}`, this.apiUrl);
+        url.searchParams.set('isNotLoginAdult', 'true');
+        const { result, data: { images, isScramble } } = await FetchJSON<APIResult<APIPages>>(this.CreateRequest(url));
+        if (result === 'ERROR') {
+            throw new Exception(R.Plugin_Common_Chapter_UnavailableError);
+        }
+
+        if (isScramble) {
+            const endpoint = new URL(`contents/images/${chapter.Parent.Identifier}/${chapter.Identifier}`, this.apiUrl);
+            const body = JSON.stringify({ line: images[0].line });
+            const { data } = await FetchJSON<APIResult<string>>(this.CreatePostRequest(endpoint, body));
+            const pages: Page<ScrambleParams>[] = [];
+            for (let image of images) {
```

Comment:
Use map instead of accumulating in loop (parallelization)

Reply:
Promises for that? fine then